### PR TITLE
fix: revert to simple icons decoration and extract the spriting logic

### PR DESCRIPTION
**Pull request by ramboz**
Most customers directly use svg icons as they were explored by their preferred application (Illustrator, etc.), and that markup isn't compatible with the spriting logic in most cases. On top of that, the current spriting implementation has required frequent patching and isn't likely that stable yet. As such, we want to revert to basic icons decoration by default, and move the spriting to the block party instead until it is considered mature enough.

Test URLs:
- Before: https://main--helix-project-boilerplate--adobe.hlx.page/
- After: https://improve-icons--helix-project-boilerplate--adobe.hlx.page/

### CODE REVIEW

File: scripts/lib-franklin.js [hunk 1]
```diff
@@ -184,26 +184,21 @@ export async function decorateIcons(element) {
...
-            styled: true,
-            html: svg
-              // rescope ids and references to avoid clashes across icons;
-              .replaceAll(/ id="([^"]+)"/g, (_, id) => ` id="${iconName}-${id}"`)
-              .replaceAll(/="url\(#([^)]+)\)"/g, (_, id) => `="url(#${iconName}-${id})"`)
-              .replaceAll(/ xlink:href="#([^"]+)"/g, (_, id) => ` xlink:href="#${iconName}-${id}"`),
-          };
-        } else {
+        // Icons using the "currentcolor" value are meant to inherit from the current
+        // page context and should be inlined
```
Comment (davidnuescheler) [inline scripts/lib-franklin.js:189]: probably should say something like...

> Icons containing "currentcolor" will be used via symbols

File: scripts/lib-franklin.js [hunk 2]
```diff
@@ -225,8 +220,10 @@ export async function decorateIcons(element) {
     const iconName = Array.from(span.classList).find((c) => c.startsWith('icon-')).substring(5);
     const parent = span.firstElementChild?.tagName === 'A' ? span.firstElementChild : span;
     // Styled icons need to be inlined as-is, while unstyled ones can leverage the sprite
```
Comment (davidnuescheler) [inline scripts/lib-franklin.js:222]: probably should say:
> `<use>` sprite or `` for non-sprited icons

**Comment by kptdobe** [comment]
We are currently exploding lib-franklin into its own library which includes adding tests... I think this refactor would be a perfect fit to add the corresponding tests to the method - See https://github.com/adobe/aem-lib/blob/main/src/decorate.js#L51. Maybe the PR should be moved there.

**Comment by kptdobe** [comment]
@ramboz [https://github.com/adobe/aem-lib is ready (only need to adjust the documentation before release). Could you move your current work in this repo ? We will be able to write corresponding tests to all the icons handling edge cases ;)

**Review by davidnuescheler — APPROVED** [review]

**Comment by ramboz** [comment]
@kptdobe Closing in favor of https://github.com/adobe/aem-lib/pull/15

**Comment by davidnuescheler** [comment]
i think it would be great to also merge the PR here... i am when we are going to switch over to `aem-lib` codebase and i just wanted to update the documentation ahead of time...

**Comment by kptdobe** [comment]
`aem-lib` is now in, the code is now in this repo.
